### ISSUE TITLE
Babel can't parse _mocha

### ISSUE THREAD

**Issue by leebyron** [douglasduteil/isparta#47]
When running

```
babel-node node_modules/.bin/isparta cover --report html node_modules/.bin/_mocha
```

As suggested by the README, I encounter this error:

```
Transformation error; return original code
{ [SyntaxError: /Users/leebyron/src/graphql-js/node_modules/mocha/bin/_mocha: 'return' outside of function (392:2)]
  pos: 9902,
  loc: { line: 392, column: 2 },
  raisedAt: 9908,
  _babel: true,
  codeFrame: '  390 |   \u001b[32m}\u001b[39m\u001b[34m\u001b[1m)\u001b[22m\u001b[39m\u001b[1m;\u001b[22m\n  391 | \n> 392 |   \u001b[36mreturn\u001b[39m\u001b[1m;\u001b[22m\n      |   ^\n  393 | \u001b[32m}\u001b[39m\n  394 | \n  395 | \u001b[90m// load\u001b[39m' }

**Comment by loganfsmyth** [comment]
Looked into this a bit since it was also reported to Babel and it seems like something has changed in `istanbul` between `0.3.14` and `0.3.15`? `isparta`s [`instrumentSync` handler](https://github.com/douglasduteil/isparta/blob/45abf56c8780414d198bc4164c08e603fc3638f1/src/cli.js#L186) is being called on the `_mocha` file.

**Comment by loganfsmyth** [comment]
Looks like mocha has merged https://github.com/mochajs/mocha/pull/1787 so this shouldn't be an issue in the future.

@leebyron Looking at this a bit more after playing with it myself, if you have your code in a subfolder, it seems setting the instrumentation root avoids this too.

### .istanbul.yml

    instrumentation:
        root: src

**Comment by leebyron** [comment]
Thanks for the investigation! This has been very helpful



—

**Comment by ScreenDriver** [comment]
Thank you @loganfsmyth . Your `.istanbul.yml` helped me out! :+1:

**Comment by Swaagie** [comment]
Still getting transpile errors with mocha@2.3 and the istanbul file does little to resolve this. 
```
➜  table git:(master) ✗ babel-node ./node_modules/.bin/isparta cover ./node_modules/.bin/mocha
Transformation error; return original code
{ [Error: Line 1: Unexpected token ILLEGAL]
  lineNumber: 1,
  description: 'Unexpected token ILLEGAL',
  index: 0 }

import TestUtils from 'react-addons-test-utils';
^^^^^^
```

**Comment by douglasduteil** [comment]
Hi @leebyron 

After some code surfing :surfer: 
I found out that the problem comes form the `istanbul` default exclusion pattern that don't match for `node_modules/.bin/_mocha`

Here is deduction http://tonicdev.com/douglasduteil/isparta-68-transformation-error-return-original-code

**Comment by jmeas** [comment]
Is this still a problem?

**Comment by chromakode** [comment]
Yes, just ran into this. Following @douglasduteil's suggestion, I changed my invocation from:

```babel-node $(npm bin)/isparta cover $(npm bin)/_mocha```

to:

```babel-node $(npm bin)/isparta cover node_modules/mocha/bin/_mocha```

and no longer get this error.

**Comment by glebmachine** [comment]
Hi everyone)

I using isparta as @chromakode said few days ago:
`./node_modules/.bin/babel-node node_modules/isparta/bin/isparta cover ./node_modules/mocha/bin/_mocha`
But nothing changes, it still throw this errors (test passed successfully, everything ok, except this erorr)

**Comment by patrickheeney** [comment]
The workaround of `node_modules/mocha/bin/_mocha` in my case will not work because my mocha lives up a directory so I must use `../node_modules/mocha/bin/_mocha` which has this same problem.

**Comment by glebmachine** [comment]
Node already support es6, just call like this:
`./node_modules/.bin/_mocha  --compilers js:babel-core/register test/index.js`

Thats all you need)

**Comment by patrickheeney** [comment]
@glebmachine Not sure I follow, my current command is: `../../node_modules/.bin/babel-node ../../node_modules/isparta/bin/isparta cover ../..//node_modules/mocha/bin/_mocha` which results in the same error you are experiencing. Am I missing something?

**Comment by glebmachine** [comment]
Sorry for misunderstanding!

Everything i ok. I'm using like u said, and everything fine:
`node_modules/.bin/babel-node node_modules/isparta/bin/isparta cover --report text --report html node_modules/mocha/bin/_mocha`.

Please, ignore my prev comment)

**Comment by niftylettuce** [comment]
It is now working with `istanbul@^1.1.0-alpha.1`.  See my setup here https://github.com/niftylettuce/koa-ratelimit-promises and my code coverage report here https://codecov.io/gh/niftylettuce/koa-ratelimit-promises.  Thanks!
